Analysis/src/TypeInfer.cpp

```diff
@@ -1803,6 +1803,8 @@ WithPredicate<TypeId> TypeChecker::checkExpr(const ScopePtr& scope, const AstExp
         result = checkExpr(scope, *a);
     else if (auto a = expr.as<AstExprIfElse>())
         result = checkExpr(scope, *a, expectedType);
+    else if (auto a = expr.as<AstExprInterpString>())
+        result = {stringType};
```

Comment:
I think this bypasses type checking of the expressions inside the string, which will hide errors in them.

Reply:
Yeah. Need to iterate over the expression and call them using `checkExpr`. We don't even need to do anything with the result, since every values can be stringified one way or another.